03-task-manager/starter/controllers/tasks.js

```diff
@@ -1,25 +1,51 @@
 const Task = require('../models/Task')
+const asyncWrapper = require('../middleware/async')  //a middleware to not repeat try catch block in every fun
+const { createCustomError } = require('../errors/custom-error')
 
-const getTasks = (req,res)=>{
-    res.send('All tasks')
-}
+//get all tasks
+const getTasks = asyncWrapper (async (req,res)=>{
+    const tasks = await Task.find({})
+    res.status(201).json({tasks})
+})
 
-const getSingleTask = (req,res)=>{
-    res.send('Get single task')
-}
+//get a task
+const getSingleTask = asyncWrapper(async(req,res)=>{
```

Comment:
Missing next as parameter
```suggestion
const getSingleTask = asyncWrapper(async(req,res,next)=>{
```